tests/plans/test_compliance.py

```diff
@@ -0,0 +1,69 @@
+import inspect
+from types import ModuleType
+from typing import Any, get_type_hints
+
+from bluesky.utils import MsgGenerator
+
+from dodal import plan_stubs, plans
+from dodal.common.types import PlanGenerator
+
+
```

Comment:
Nit: This is sort of a structural/architectural test rather than a unit test. I think it could do with a docstring at the top preparing the reader for that fact and explaining it's general perpose